### DISCUSSION TITLE
Changes remain stuck on 'Error' Status after hitting 'Apply'

### DISCUSSION THREAD

**Discussion by kcredit** [opengisch/QField#3986] [Bugs]
I have a Community project for one of the classes that I am teaching this semester called 'GY208_CampusGreenSpaceMapping' in which my students have been assigned to groups to add tree locations (and a few attributes) around our campus using QField. While many of the initial additions to our 'Trees_Test' layer synched with no problems, recent additions to the 'Trees_Test' and specific group layers (e.g., 'Trees_1', 'Trees_2', 'Trees_3', 'Trees_4') have not synched. As the owner of the organisation and project, I can observe all of the students' attempted changes in the 'Changes' panel of the QFieldCloud site. However, their status is listed as 'Error' (with no additional description or documentation), as shown here:

![Screenshot 2023-02-28 at 16 31 23](https://user-images.githubusercontent.com/31995630/221916723-81fa6407-a688-4ba2-8514-94c0e612097d.png)

Each individual change is also strangely listed as 'Geometry not modified' even though they are, in fact, changing the geometry of the layer by adding points:

![Screenshot 2023-02-28 at 16 36 34](https://user-images.githubusercontent.com/31995630/221918065-221daad1-e1bb-463a-82d4-d00495c15792.png)

I can, of course, choose the 'Apply' action for each of these (thousands) of changes, but each time I do that, the Status changes to 'Started' or 'Pending' before eventually going back to 'Error' with no additional documentation or information on the source or reason for the error. I'm not sure what the issue is, because these layers were working fine a few weeks ago. It would be useful to simply be able to download these changes in bulk in a new shapefile (even if resolving errors within the existing layer isn't possible) so that they have something to map.

**Comment by suricactus** [comment]
Actually there is more information, if you go to click on "See jobs" in the last screenshot, you can see all attempts that QFieldCloud made to apply your changes. This is already in our documentation https://docs.qfield.org/reference/qfieldcloud/jobs/?h=logs#troubleshoot_2 .


 Get the last one and if you don't understand the feedback, feel free to ask more.

**Reply by kcredit** [reply]
OK great, that is helpful - so going into each of the 'Delta Apply' details for the failed jobs, I'm seeing a couple of different issues that aren't mentioned in the documentation (as far as I can see)?

Most of them have a failure on 'Upload Project':

![Screenshot 2023-03-01 at 16 33 42](https://user-images.githubusercontent.com/31995630/222202851-84af0457-a38c-4f17-9cab-05e409d87429.png)

I also saw this error in one:

![Screenshot 2023-03-01 at 16 34 26](https://user-images.githubusercontent.com/31995630/222203043-1469ee49-6cb4-4ec4-9312-8dc645d76c8f.png)

How do we fix the failed upload?

**Comment by jescott2** [comment]
I am running into a similar issue. (QGIS Ver 3.28.4-Firenze, QField  Sync ver 4.4.9, Windows 10)

Apparently the Delta_Apply worked ( green check) but Upload did not work (red check). I ran a synchronize from my iPad. Before the sync, my three new points in utilities appeared on my iPad screen. Now the points are gone but I hope the points are still available in the system. I have read the Troubleshooting linked above but I don't see notes on my case. I have also tried Applying the job again. The Job Status changes to Pending, then Started, then Error. I pasted the expanded printout for teh Delata_Ap-ly and Upload steps below.

Can someone help me get back my 3 new points for the Utilities layer?

![image](https://user-images.githubusercontent.com/38323329/233187060-0a253cd7-996e-4428-8ef8-f71e8de9a0d2.png)

Delta Apply (13a4f69c-84ad-4941-956a-0e586f7449d7)
×
Start QGIS Application
Download Project Directory
17:36:35.408 root      INFO     Downloading file "20210-PRELIM(12-22-20)_modified.tif"…
17:36:40.265 root      INFO     Downloading file "Floorplanver12.gpkg"…
17:36:42.219 root      INFO     Downloading file "NewHome_cloud_cloud.qgs"…
17:36:43.439 root      INFO     Downloading file "NewHome_cloud_cloud_attachments.zip"…
17:36:44.995 root      INFO     Downloading file "Ortho 2019.tif"…
17:36:46.270 root      INFO     Downloading file "PropertyFiles.gpkg"…
17:36:48.665 root      INFO     Downloading file "Slope Byte.tif"…
17:36:48.946 root      INFO     Local files list for project "5a677fb2-c84e-418f-86ba-d2175a3a2cb0":
Name                                     Size  MD5 Checksum
-----------------------------------  --------  --------------------------------
20210-PRELIM(12-22-20)_modified.tif  58385201  067cd3e0924a3c8a7bf790398e15889a
Floorplanver12.gpkg                   1224704  9aa8da1cecb3361e54ec0d08978eec84
NewHome_cloud_cloud.qgs                584378  096f4dedd20a59e9dcba81ad15aa119c
NewHome_cloud_cloud_attachments.zip      1126  88be6972047554d76c27655b3de7965a
Ortho 2019.tif                        3388238  f200b1da5fc577e1b9c1a61d9e839161
PropertyFiles.gpkg                   25329664  46b82134713163197661558ab65b1f96
Slope Byte.tif                          91193  8587c9314d421548bce8a26d22094681
Apply Deltas
17:36:48.947 root      INFO     Loading project file "/tmp/tmp2q2972_z/files/NewHome_cloud_cloud.qgs"...
17:36:48.998 QGSSTDERR WARNING  Logged warning: Loading a file that was saved with a newer version of qgis (saved in 3.28.4-Firenze, loaded in 3.28.3-Firenze). Problems may occur.
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: proj=hgridshift (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: proj=hgridshift (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: proj=hgridshift (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: proj=hgridshift (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: inv (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: inv (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: inv (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: inv (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: proj=hgridshift (File not found or invalid)
proj_create: Error 1029 (File not found or invalid): pipeline: Pipeline: Bad step definition: inv (File not found or invalid)
17:36:49.919 root      INFO     Loading delta file "/io/deltafile.json"...
17:36:50.028 qfieldcloud.qgis.apply_deltas INFO     Successfully applied delta on layer "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83"
17:36:50.030 qfieldcloud.qgis.apply_deltas WARNING  The modified feature pk valued does not match "sourcePk" in the delta in "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83": sourcePk= modifiedFeaturePk=6
17:36:50.077 qfieldcloud.qgis.apply_deltas INFO     Successfully applied delta on layer "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83"
17:36:50.077 qfieldcloud.qgis.apply_deltas WARNING  The modified feature pk valued does not match "sourcePk" in the delta in "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83": sourcePk= modifiedFeaturePk=7
17:36:50.132 qfieldcloud.qgis.apply_deltas INFO     Successfully applied delta on layer "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83"
17:36:50.132 qfieldcloud.qgis.apply_deltas WARNING  The modified feature pk valued does not match "sourcePk" in the delta in "Utilities_25862eca_ed40_46dd_846e_36fbba0efd83": sourcePk= modifiedFeaturePk=8
Stop QGIS Application
Upload Project
17:36:51.559 root      INFO     Local files list for project "5a677fb2-c84e-418f-86ba-d2175a3a2cb0":
Name                                     Size  MD5 Checksum
-----------------------------------  --------  --------------------------------
20210-PRELIM(12-22-20)_modified.tif  58385201  067cd3e0924a3c8a7bf790398e15889a
Floorplanver12.gpkg                   1224704  2d52450a670928e9001f0774c3965112
NewHome_cloud_cloud.qgs                584378  096f4dedd20a59e9dcba81ad15aa119c
NewHome_cloud_cloud_attachments.zip      1126  88be6972047554d76c27655b3de7965a
Ortho 2019.tif                        3388238  f200b1da5fc577e1b9c1a61d9e839161
PropertyFiles.gpkg                   25329664  1bb79e87504ae0e63e49c56fdef5985a
Slope Byte.tif                          91193  8587c9314d421548bce8a26d22094681
17:36:52.903 root      INFO     Uploading file "Floorplanver12.gpkg"…
17:36:55.641 root      INFO     Uploading file "PropertyFiles.gpkg"…
